pyop2/codegen/builder.py

```diff
@@ -371,14 +373,23 @@ def pack(self, loop_indices=None):
             shape = (1, )
         rshape = shape + rmap.shape[1:] + (rdim, )
         cshape = shape + cmap.shape[1:] + (cdim, )
+        return (rshape, cshape)
```

Comment:
What information is encoded in rmap.shape[1:]? I am trying to understand why you multiply these values together in line 486.

Reply:
`rmap.shape` is the shape of the map from the iteration set into the values at each entry of the set. So it's for example `(10, 3)` for a cell_node_map for P1 on 10 triangles. But really it's `(None, 3)` because the first entry is runtime known. So `rmap.shape[1:]` is the number of dofs that the kernel will see for this argument on the current set entity.